### ISSUE TITLE
[firebase_auth] Firebase SMS Authentication Error Code 39

### ISSUE THREAD

**Issue by mmd44** [firebase/firebase-android-sdk#5815]
### Versions
  * Android Studio version: Jellyfish | 2023.3.2 Canary 1
  * Firebase Component: firebase-auth 
  * Component version: BOM: 32.8.0

### Description
I am facing a weird issue with firebase sms auth in Android and Jetpack compose.

I am trying to test **reCAPTCHA** flow when Play Integrity Check is not available on REAL NUMBERS (when testing with fictional numbers, everything works fine).

I have an app that is **not** yet on the store, so assuming Play Integrity API will fail and firebase auth sdk will try to validate my app through ReCaptcha as stated in docs.

I have already setup the following:

- Enabled reCAPTCHA Entreprise on Google console
- Added debug and release signing keys SHA1 and SHA256 to the firebase console
- No country restriction is on firebase console
- API Key on google console is not constrained
- My app domain name (xxxxxx.firebaseapp.com) is added to firebase's authorized domains
- Tested on both release and debug versions of my app but got same result

### Steps to reproduce:

The **weird** behavior is as following:

I have 2 real phone numbers one with country code +1 (CAD) and the other +961 (LB) (both are listed as supported countries);

**1.** When I test with **+1 XXX XXX XXXX**:

I **successfully** receive the code while noticing the following errors in **logs**:

- [SmsRetrieverHelper] SMS verification code request failed: unknown status code: 18002 Invalid PlayIntegrity token; app not Recognized by Play Store.
- Failed to initialize reCAPTCHA config: No Recaptcha Enterprise siteKey configured for tenant/project *

**2.** When I test with **+961 XX XXXXXX**:

I **fail** to receive the code, get the **same errors in logs as above** + a **failure message**:
- [SmsRetrieverHelper] SMS verification code request failed: unknown status code: 17499 Error code:39

In both test cases, the app successfully opens the browser (a screen with 'verifying you're not a robot' label text and loading) and redirects back to my app, so I am assuming the reCAPTCHA is working...

I have also tried to add the recaptcha sdk and linking it to a key added on Google console and initialize it on my app but I also failed to get any different result. 

Any idea what could be the issue?

**Comment by Xiaoshouzi-gh** [comment]
We use recaptcha V2 instead of Entreprise internally with phone auth, can you reproduce both cases after you Disable reCAPTCHA Entreprise on Google console?

**Comment by mmd44** [comment]
Yes same results.
It seems reCAPTCHA Enterprise is not being used even when enabled as no requests are being registered on this API on Google console's enabled APIs page.

**Comment by nxnjz** [comment]
This is also an issue for us with +961 numbers.

**Comment by MDikkii** [comment]
It has similar issue with Sierra Leonian numbers...

**Comment by kwagdy** [comment]
Same for "+2011..." numbers

**Comment by neobie** [comment]
Having same problem

**Comment by yehya** [comment]
Yeah firebase has banned all (+20 Egypt) from using firebase phone auth and this is not documented anywhere. Anyone from the firebase team can comment or find out what is going on? I even had this issue when signing up for google play & it wouldn't accept my egyptian number for verification.

**Comment by yehya** [comment]
Egypt is on the list of supported regions so very confused.
https://firebase.google.com/support/faq#develop

**Comment by lehcar09** [comment]
Hi folks, sorry for the silence here. I took a look at this issue and I believe this is not an SDK issue. 

To ensure the safety and reliability of our services, carriers or entire regions with consistently low SMS success rates may become temporarily unavailable. This has been implemented as part of the automated security systems that protect Firebase Phone Authentication customers from unexpected billing from carriers with very low success rates. 

To get better help, it would be best to reach out to the [Firebase support channel ](https://firebase.google.com/support/)instead. We'd need access to your project settings (including projectID and private information) to troubleshoot this type of issue.
